src/main/java/com/google/devtools/build/lib/events/Reporter.java

```diff
@@ -39,7 +39,7 @@
  * <p>Thread-safe: calls to {@code #report} may be made on any thread. Handlers may be run in an
  * arbitrary thread (but right now, they will not be run concurrently).
  */
-public final class Reporter implements ExtendedEventHandler {
```

Comment:
Rather than making Reporter non-final for mocking purposes, can't you use the [EventCollector](src/main/java/com/google/devtools/build/lib/events/EventCollector.java) as in for instance [FoundationTestCase.java](src/test/java/com/google/devtools/build/lib/testutil/FoundationTestCase.java)?